src/OpenTelemetry/Metrics/BaseExportingMetricReader.cs

```diff
@@ -20,24 +20,21 @@ namespace OpenTelemetry.Metrics
 {
     public class BaseExportingMetricReader : MetricReader
     {
-        private readonly BaseExporter<Metric> exporter;
-        private bool disposed;
+        protected readonly BaseExporter<Metric> exporter;
```

Comment:
This would probably generate a warning if we had static analysis turned on. Generally, don't want to expose fields. Easy fix to make them into properties:

```csharp
protected BaseExporter<Metric> Exporter { get; }
protected bool IsDisposed { get; private set; } // Really shouldn't be exposed at all should it?
```

Reply:
The current `disposed` is definitely very messy (I was chatting with @cijothomas) yesterday.
Let me clean this up in the next PR.

Reply:
Something about the dispose pattern in these classes looks off to me. I feel like `bool disposed` shouldn't leak out of the base class. Thinking something like this...

```csharp
    public abstract class MetricReader : IDisposable
    {
        private bool disposed;

        ~MetricReader()
        {
            if (!this.disposed)
            {
                this.Dispose(false);
            }
        }

        public void Dispose()
        {
            if (!this.disposed)
            {
                this.Dispose(true);
                GC.SuppressFinalize(this);
                this.disposed = true;
            }
        }

        protected virtual void Dispose(bool isDisposing)
        {
        }
    }
```

Reply:
#2355

Reply:
Yeah, this was introduced in #2332. I think the simple answer is to redo the work like what we've done here https://github.com/open-telemetry/opentelemetry-dotnet/blob/3cc0f7e4af03a49942b4e2b75d383b73b91e2ad6/src/OpenTelemetry/BaseExportProcessor.cs#L30.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/2357/files#r709698431